test/test_bessel_j.hpp

```diff
@@ -281,7 +281,10 @@ void test_bessel(T, const char* name)
     BOOST_CHECK_EQUAL(boost::math::cyl_bessel_j(T(0), T(2.5)), boost::math::cyl_bessel_j(T(0), T(-2.5)));
     BOOST_CHECK_EQUAL(boost::math::cyl_bessel_j(T(1), T(2.5)), -boost::math::cyl_bessel_j(T(1), T(-2.5)));
     #ifndef SYCL_LANGUAGE_VERSION
-    BOOST_CHECK_CLOSE_FRACTION(boost::math::cyl_bessel_j(364, T(38.5)), SC_(1.793940496519190500748409872348034004417458734118663909894e-309), tolerance);
+    BOOST_IF_CONSTEXPR (std::numeric_limits<T>::min_exponent10 <= -307)
```

Comment:
Denormals not withstanding, this should work for all T as well - it should underflow to zero when the exponent does not have enough range?